community_images/httpd/bitnami/run.sh

```diff
@@ -0,0 +1,133 @@
+#!/bin/bash
+
+set -x
+set -e
+
+# shellcheck disable=SC1091
+SCRIPTPATH="$( cd -- "$(dirname "$0")" >/dev/null 2>&1 ; pwd -P )"
+
+# shellcheck disable=SC1091
+. "${SCRIPTPATH}"/../../common/helpers.sh
+
+
+INPUT_REGISTRY=docker.io
+INPUT_ACCOUNT=bitnami
+REPOSITORY=apache
+
+if [ "$#" -ne 1 ]; then
+    PUBLISH_IMAGE="no"
+else
+    PUBLISH_IMAGE=$1
+fi
+
+test()
+{
+    local IMAGE_REPOSITORY=$1
+    local TAG=$2
+    local NAMESPACE=$3
+    local HELM_RELEASE="$REPOSITORY"-release
+
+    echo "Testing $REPOSITORY"
+
+    # upgrade helm
+    helm repo update
+
+    # Install helm
+    with_backoff helm install "${HELM_RELEASE}" "${INPUT_ACCOUNT}"/"${REPOSITORY}" --namespace "${NAMESPACE}" --set image.tag="${TAG}" --set image.repository="${IMAGE_REPOSITORY}" -f "${SCRIPTPATH}"/overrides.yml
+    report_pulls "${IMAGE_REPOSITORY}"
+
+    # waiting for pod to be ready
+    echo "waiting for pod to be ready"
+    kubectl wait deployments "${HELM_RELEASE}" -n "${NAMESPACE}" --for=condition=Available=True --timeout=10m
+
+    # get pod name
+    POD_NAME=$(kubectl -n "${NAMESPACE}" get pods -l app.kubernetes.io/name="$REPOSITORY" -o jsonpath="{.items[0].metadata.name}")
+
+    # fetch service url and store the urls in URLS file
+    rm -f URLS
+    minikube service "${HELM_RELEASE}" -n "${NAMESPACE}" --url >> URLS
+
+    # create ssl certs
+    cleanup_certs
+    create_certs
+
+    # Changing "http" to "https" in the urls file
+    sed -i '2,2s/http/https/' URLS
+    cat URLS
+
+    # curl to urls
+    while read p;
+    do
+        curl -k "${p}"
+    done <URLS
+
+    #Removing urls file
+    rm URLS
+
+    # fetch minikube ip
+    MINIKUBE_IP=$(minikube ip)
+
+    # curl to https url
+    curl http://"${MINIKUBE_IP}" -k
+
+    # copy common_commands.sh into container
+    kubectl -n "${NAMESPACE}" cp "${SCRIPTPATH}"/../../common/tests/common_commands.sh "${POD_NAME}":/tmp/common_commands.sh
+
+    # run command on cluster
+    kubectl -n "${NAMESPACE}" exec -i "${POD_NAME}" -c apache -- /bin/bash -c "/tmp/common_commands.sh"
+
+    # bring down helm install
+    helm delete "${HELM_RELEASE}" --namespace "${NAMESPACE}"
+
+    # delete the PVC associated
+    kubectl -n "${NAMESPACE}" delete pvc --all
+
+    # update image in docker-compose yml
+    sed "s#@IMAGE#${IMAGE_REPOSITORY}:${TAG}#g" "${SCRIPTPATH}"/docker-compose.yml.base > "${SCRIPTPATH}"/docker-compose.yml
+
+    # install docker container
+    docker-compose -f "${SCRIPTPATH}"/docker-compose.yml -p "${NAMESPACE}" up -d
+    report_pulls "${IMAGE_REPOSITORY}"
+
+    # sleep for 30 sec
+    sleep 30
+
+    # exec into container and run coverage script
+    pwd
+    chmod 777 "${SCRIPTPATH}"/coverage_script.sh
```

Comment:
The permissions can be changed outside once in the repo and this line can be removed

Reply:
Agreed, will do that.